src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -198,22 +357,36 @@ const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({ submiss
 
         <Footer>
           <Flex justifyContent="justify-center" spaceX="space-x-3">
+            <Button
+              size="xs"
+              icon={() => <BpIcon icon={IconNames.DOUBLE_CHEVRON_LEFT} />}
+              variant="light"
+              onClick={() => resetPage()}
+              disabled={page <= 0}
+            />
             <Button
               size="xs"
               icon={() => <BpIcon icon={IconNames.ARROW_LEFT} />}
               variant="light"
-              onClick={() => table.previousPage()}
-              disabled={!table.getCanPreviousPage()}
+              onClick={() => setPage(page - 1)}
```

Comment:
You should ideally use an updater function for setting state to values that rely on the previous state. This minimises the possibility of bugs like multiple state updates getting uncaught:

```suggestion
              onClick={() => setPage(p => p - 1)}
```